pkg/monitoring/rules/alerts/prometheus.go

```diff
@@ -0,0 +1,55 @@
+package alerts
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+)
+
+const (
+	prometheusRunbookAnnotationKey = "runbook_url"
+	partOfAlertLabelKey            = "kubernetes_operator_part_of"
+	partOfAlertLabelValue          = "kubevirt"
+	componentAlertLabelKey         = "kubernetes_operator_component"
+	componentAlertLabelValue       = "ssp-operator"
+	defaultRunbookURLTemplate      = "https://kubevirt.io/monitoring/runbooks/%s"
+	runbookURLTemplateEnv          = "RUNBOOK_URL_TEMPLATE"
+)
+
+func Register() error {
+	alerts := [][]promv1.Rule{
+		operatorAlerts(),
+	}
+
+	runbookURLTemplate, err := getRunbookURLTemplate()
+	if err != nil {
+		return err
+	}
+
+	for _, alertGroup := range alerts {
+		for _, alert := range alertGroup {
+			alert.Labels[partOfAlertLabelKey] = partOfAlertLabelValue
+			alert.Labels[componentAlertLabelKey] = componentAlertLabelValue
+			alert.Annotations[prometheusRunbookAnnotationKey] = fmt.Sprintf(runbookURLTemplate, alert.Alert)
```

Comment:
In my opinion, it is not a good design to modify the variable `alert` in the loop body, because it is a shallow copy of the element in the `alertGroup` slice. In general, modifying the copy will not modify the original element in the slice.

Can you please use a pointer to the element instead of a copy?
```go
for i := range alertGroup {
	alert := &alertGroup[i]
	alert.Labels[partOfAlertLabelKey] = partOfAlertLabelValue
	alert.Labels[componentAlertLabelKey] = componentAlertLabelValue
	alert.Annotations[prometheusRunbookAnnotationKey] = fmt.Sprintf(runbookURLTemplate, alert.Alert)
}
```

Reply:
This is how we do it on all operators.. is it important for you to fix? Just want to align with all

Reply:
Yes, it's important. The current code is not correct, it just works "by accident", because you are modifying only fields `Labels` and `Annotations`, which are maps. If you were to modify a non-pointer field, like `alert.Recort` then it would not work.

Reply:
Just a note: we did it this way in all other cnv operators (hco, kubevirt, cnao, hpp, cdi) so I would prefer to align with all but if it's a bug it should be handled.

Reply:
I don't like it, but it's not important, because it is not a bug.
If @0xFelix is ok with it, we can keep the incorrect version.

Reply:
I would prefer to straighten this out right away. We could make this the first repository to show how this can be improved in all other repos. :)

Reply:
fixed thank you!